### DISCUSSION TITLE
Strange POWER command happening on MQTT connect

### DISCUSSION THREAD

**Discussion by cdntinker** [arendst/Tasmota#22679] [Support request] [answered]
### PROBLEM DESCRIPTION
_A clear and concise description of what the problem is._

### REQUESTED INFORMATION
_Make sure your have performed every step and checked the applicable boxes before submitting your issue. Thank you!_

- [x] Read the [Contributing Guide and Policy](https://github.com/arendst/Tasmota/blob/development/CONTRIBUTING.md) and [the Code of Conduct](https://github.com/arendst/Tasmota/blob/development/CODE_OF_CONDUCT.md)
- [x] Searched the problem in [issues](https://github.com/arendst/Tasmota/issues)
- [x] Searched the problem in [discussions](https://github.com/arendst/Tasmota/discussions)
- [x] Searched the problem in the [docs](https://tasmota.github.io/docs/FAQ)
- [ ] Searched the problem in the [chat](https://discord.gg/Ks2Kzd4)
- [x] Device used (e.g., Sonoff Basic): Sonoff Dual R2
- [x] Tasmota binary firmware version number used: 14.3.0
  - [x] Pre-compiled
  - [ ] Self-compiled
- [x] Flashing tools used: OTA upgrade
- [x] Provide the output of command: `Backlog Template; Module; GPIO 255`:
```lua
  Configuration output here:
21:56:48.631 CMD: Backlog Template; Module; GPIO 255
21:56:48.662 MQT: stat/AquariumLights/RESULT = {"NAME":"Generic","GPIO":[1,1,1,1,1,1,1,1,1,1,1,1,1,1],"FLAG":0,"BASE":18}
21:56:48.884 MQT: stat/AquariumLights/RESULT = {"Module":{"39":"Sonoff Dual R2"}}
21:56:49.135 MQT: stat/AquariumLights/RESULT = {"GPIO0":{"0":"None"},"GPIO1":{"0":"None"},"GPIO2":{"0":"None"},"GPIO3":{"0":"None"},"GPIO4":{"0":"None"},"GPIO5":{"225":"Relay2"},"GPIO9":{"0":"None"},"GPIO10":{"32":"Button1"},"GPIO12":{"224":"Relay1"},"GPIO13":{"320":"Led_i1"},"GPIO14":{"0":"None"},"GPIO15":{"0":"None"},"GPIO16":{"0":"None"},"GPIO17":{"0":"None"}}

```
- [ ] If using rules, provide the output of this command: `Backlog Rule1; Rule2; Rule3`:
```lua
  Rules output here:


```
- [x] Provide the output of this command: `Status 0`:
```lua
  STATUS 0 output here:
21:59:37.106 CMD: Status 0
21:59:37.114 RSL: STATUS = {"Status":{"Module":39,"DeviceName":"Aquarium Lights","FriendlyName":["Aquarium_Lights_1","Aquarium_Lights_2"],"Topic":"AquariumLights","ButtonTopic":"0","Power":"01","PowerLock":"00","PowerOnState":3,"LedState":1,"LedMask":"FFFF","SaveData":1,"SaveState":1,"SwitchTopic":"0","SwitchMode":[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],"ButtonRetain":0,"SwitchRetain":0,"SensorRetain":0,"PowerRetain":0,"InfoRetain":0,"StateRetain":0,"StatusRetain":0}}
21:59:37.119 RSL: STATUS1 = {"StatusPRM":{"Baudrate":115200,"SerialConfig":"8N1","GroupTopic":"tasmotas","OtaUrl":"http://ota.tasmota.com/tasmota/release/tasmota.bin.gz","RestartReason":"Software/System restart","Uptime":"0T00:13:59","StartupUTC":"2024-12-18T20:45:38","Sleep":50,"CfgHolder":4617,"BootCount":35,"BCResetTime":"2024-11-10T19:49:10","SaveCount":982,"SaveAddress":"F5000"}}
21:59:37.123 RSL: STATUS2 = {"StatusFWR":{"Version":"14.3.0(release-tasmota)","BuildDateTime":"2024-10-15T08:18:01","Boot":31,"Core":"2_7_7","SDK":"2.2.2-dev(38a443e)","CpuFrequency":80,"Hardware":"ESP8285N08","CR":"413/699"}}
21:59:37.127 RSL: STATUS3 = {"StatusLOG":{"SerialLog":2,"WebLog":2,"MqttLog":0,"SysLog":0,"LogHost":"","LogPort":514,"SSId":["VictoriaDrive2B","VictoriaDrive3"],"TelePeriod":300,"Resolution":"558180C0","SetOption":["00008009","2805C80001000600003C5A0A192800000000","00000080","00006000","00004000","00000000"]}}
21:59:37.138 RSL: STATUS4 = {"StatusMEM":{"ProgramSize":649,"Free":352,"Heap":17,"ProgramFlashSize":1024,"FlashSize":1024,"FlashChipId":"144051","FlashFrequency":40,"FlashMode":"DOUT","Features":["0809","8F9AC787","04368001","000000CF","010013C0","C000F981","00004004","00001000","54000020","00000080","00000000"],"Drivers":"1,2,!3,!4,!5,!6,7,!8,9,10,12,!16,!18,!19,!20,!21,!22,!24,26,!27,29,!30,!35,!37,!45,62,!68","Sensors":"1,2,3,4,5,6","I2CDriver":"7"}}
21:59:37.143 RSL: STATUS5 = {"StatusNET":{"Hostname":"AquariumLights-2458","IPAddress":"192.168.9.27","Gateway":"192.168.8.1","Subnetmask":"255.255.252.0","DNSServer1":"192.168.8.1","DNSServer2":"127.0.0.1","Mac":"60:01:94:AC:89:9A","Webserver":2,"HTTP_API":1,"WifiConfig":4,"WifiPower":17.0}}
21:59:37.146 RSL: STATUS6 = {"StatusMQT":{"MqttHost":"mosquitto1","MqttPort":1883,"MqttClientMask":"DVES_%06X","MqttClient":"DVES_AC899A","MqttUser":"DVES_USER","MqttCount":1,"MAX_PACKET_SIZE":1200,"KEEPALIVE":30,"SOCKET_TIMEOUT":4}}
21:59:37.151 RSL: STATUS7 = {"StatusTIM":{"UTC":"2024-12-18T20:59:37Z","Local":"2024-12-18T21:59:37","StartDST":"2024-03-31T02:00:00","EndDST":"2024-10-27T03:00:00","Timezone":"+01:00","Sunrise":"08:39","Sunset":"16:54"}}
21:59:37.156 RSL: STATUS10 = {"StatusSNS":{"Time":"2024-12-18T21:59:37"}}
21:59:37.161 RSL: STATUS11 = {"StatusSTS":{"Time":"2024-12-18T21:59:37","Uptime":"0T00:13:59","UptimeSec":839,"Heap":17,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":1,"POWER1":"ON","POWER2":"OFF","Wifi":{"AP":1,"SSId":"VictoriaDrive2B","BSSId":"94:83:C4:06:74:F9","Channel":5,"Mode":"11n","RSSI":70,"Signal":-65,"LinkCount":1,"Downtime":"0T00:00:05"}}}

```
- [ ] Set `weblog` to 4 and then, when you experience your issue, provide the output of the Console log:
```lua
  Console output here:


```

### TO REPRODUCE
_Steps to reproduce the behavior:_

### EXPECTED BEHAVIOUR
_A clear and concise description of what you expected to happen._

I expect the device to NOT toggle its own states without being told to do so.

### SCREENSHOTS
_If applicable, add screenshots to help explain your problem._

### ADDITIONAL CONTEXT
_Add any other context about the problem here._

**(Please, remember to close the issue when the problem has been addressed)**

This is a weird one...

I have a Sonoff Dual R2 running lights on an aquarium.  Lately, there has been a lot of WiFi interferance and now, every time it reconnects to my MQTT broker, it seems to be sending out a null message on "cmnd/AquariumLights/POWER".

When this happens, if the lights are in night mode (POWER1 = off, POWER2 = on), it toggles to day mode (POWER1 = on, POWER2 = off). Even stranger, this has no effect if the device is already in day mode.

I have been completely unable, through MANY hours of research, to find any sort of reference to this behaviour.

Is this in any way normal?

Is there any way to make it stop?

**Comment by sfromis** [comment]
You "forgot" to provide the log for when you see what surprised you. That's one of the most important parts of the form to fil in.

Anyway, the the empty "cmnd/AquariumLights/POWER" is not actually a command it is processing, instead (for backwards compatibility) a message originally allowing other subscribers to see it being available.

Whatever you see with toggling is unrelated to that, but might depend on other details from the detailed log you did not provide.

**Reply by cdntinker** [reply]
It is actually SENDING the empty "cmnd/AquariumLights/POWER".  I determined this by pointing it at its very own MQTT server.  Nothing else talking to it except a "mosquitto_sub" in a terminal window to track what's being communicated.

As for the log...

What I'm seeing in the web cosole is just:
```
14:39:41.491 MQT: Connect failed to mosquitto1:1883, rc -2. Retry in 20 sec
14:40:02.603 MQT: Attempting connection...
14:40:02.650 MQT: Connected
14:40:02.654 MQT: tele/AquariumLights/LWT = Online (retained)
14:40:02.657 MQT: cmnd/AquariumLights/POWER = 
14:40:02.873 MQT: stat/AquariumLights/RESULT = {"POWER1":"ON"}
14:40:02.877 MQT: stat/AquariumLights/POWER1 = ON
14:40:02.921 MQT: stat/AquariumLights/RESULT = {"POWER2":"OFF"}
14:40:02.926 MQT: stat/AquariumLights/POWER2 = OFF
```
every time it happens.  Often with widely varying numbers of "MQT: Connect failed" cycles.

So far, "WebLog 4" just gives me:
```
18:41:16.032 CMD: weblog 4
18:41:16.039 MQT: stat/AquariumLights/RESULT = {"WebLog":4}
18:41:16.399 CFG: Saved to flash at F9, Count 1058, Bytes 4096
18:41:22.791 WIF: Checking connection...
18:41:37.783 WIF: Sending Gratuitous ARP
18:41:42.744 WIF: Checking connection...
18:42:02.752 WIF: Checking connection...
18:42:23.539 WIF: Checking connection...
18:42:37.882 WIF: Sending Gratuitous ARP
18:42:43.763 WIF: Checking connection...
18:43:03.990 WIF: Checking connection...
18:43:25.001 WIF: Checking connection...
18:43:37.819 WIF: Sending Gratuitous ARP
18:43:45.517 WIF: Checking connection...
18:44:05.501 WIF: Checking connection...
18:44:16.790 MQT: tele/AquariumLights/STATE = {"Time":"2024-12-19T18:44:16","Uptime":"0T20:58:38","UptimeSec":75518,"Heap":24,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":54,"POWER1":"ON","POWER2":"OFF","Wifi":{"AP":1,"SSId":"VictoriaDrive2B","BSSId":"94:83:C4:06:74:F9","Channel":5,"Mode":"11n","RSSI":68,"Signal":-66,"LinkCount":23,"Downtime":"0T00:11:39"}}
18:44:25.522 WIF: Checking connection...
18:44:37.794 WIF: Sending Gratuitous ARP
18:44:45.499 WIF: Checking connection...
18:45:05.762 WIF: Checking connection...
18:45:26.342 WIF: Checking connection...
18:45:37.812 WIF: Sending Gratuitous ARP
```
with those last 4 lines repeated continiously & the tele/ line interspersed.

if it ever shows something else while I'm looking (or that I can scroll back & find in the multitude of lines output...) I'll come back & post it...

**Reply by sfromis** [reply]
> It is actually SENDING the empty "cmnd/AquariumLights/POWER"

Yes, but it does not matter. Not at all. Unless some other MQTT client is subscribing to such topics, which would be odd. MQTT connect trouble is happening outside of Tasmota, you need to debug your setup if you want MQTT to work at that time.

Logs at the time of incident (like the toggling you mentioned) would be interesting, not just turning on logging at a random time where *nothing* is happening. Those `WIF` messages are just normal operation, not indicative of anything except that network connection works.

**Reply by cdntinker** [reply]
I find the device itself sending a "cmnd" very odd.  

As I said in the last message, The unwanted toggling (Always immediately after it sends that cmnd) happens even when the only other thing even talking to the broker is a mosquitto_sub command running in a terminal. So it's definitely NOT something else toggling it.

As a test, I've sent 
```
mosquitto_pub -h mosquitto1 -t cmnd/AquariumLights/POWER -m ""
```
& it toggles.  But again, only when it's in night mode...

Yes, the connection problem is definitely outside Tasmota itself.  I've been battling outside interference for a while.  2.4 GHz WiFi above ground level is getting slammed by something.  I've almost come to terms with this, frustrating as it is.

Having the aquarium lights toggle when they're supposed to be in night mode, on the other hand, is intensely frustrating.

**Reply by cdntinker** [reply]
Well...  It finally did the reconnect thing...
```
20:31:17.905 WIF: Network (re)scan started...
20:31:20.896 WIF: Network 0, AP-, SSId BELL609, Channel 1, BSSId 98:42:65:F6:BB:1E, RSSI -81, Encryption 1
20:31:20.899 WIF: Network 1, AP-, SSId CeciliasWeb, Channel 1, BSSId 8E:6A:8D:6F:03:89, RSSI -83, Encryption 1
20:31:20.902 WIF: Network 2, AP-, SSId TinkerToys, Channel 1, BSSId 94:83:C4:06:75:43, RSSI -82, Encryption 1
20:31:20.904 WIF: Network 3, AP-, SSId VictoriaDrive1, Channel 2, BSSId 9C:A2:F4:2F:4E:26, RSSI -72, Encryption 1
20:31:20.908 WIF: Network 4, AP-, SSId IoTicleing, Channel 2, BSSId 1C:61:B4:6B:CD:28, RSSI -60, Encryption 1
20:31:20.912 WIF: Network 5, AP-, SSId IoTesting, Channel 3, BSSId 94:83:C4:06:75:77, RSSI -82, Encryption 1
20:31:20.914 WIF: Network 6, AP1, SSId VictoriaDrive2B, Channel 5, BSSId 94:83:C4:06:74:F9, RSSI -60, Encryption 1
20:31:20.917 WIF: Network 7, AP-, SSId Covrd Mesh, Channel 6, BSSId EC:AD:E0:3D:5F:41, RSSI -82, Encryption 1
20:31:20.920 WIF: Network 8, AP-, SSId ARLO_VMB_1684928551, Channel 6, BSSId 9C:3D:CF:49:AB:AC, RSSI -80, Encryption 1
20:31:20.921 WIF: Network 9, AP2, SSId VictoriaDrive3, Channel 7, BSSId 1C:61:B4:6B:CC:F6, RSSI -48, Encryption 1
20:31:20.923 WIF: Network 10, AP-, SSId LF_925_0f9c, Channel 9, BSSId A6:97:5C:9A:0F:9C, RSSI -85, Encryption 1
20:31:20.925 WIF: Network 11, AP-, SSId BELL005, Channel 11, BSSId 34:8A:AE:69:26:4E, RSSI -88, Encryption 1
20:31:22.001 WIF: Connecting to AP2 VictoriaDrive3 Channel 7 BSSId 1C:61:B4:6B:CC:F6 in mode 11n as AquariumLights-2458...
20:31:22.006 MQT: Attempting connection...
20:31:23.009 WIF: DNS failed for mosquitto1 after 1001 ms
20:31:23.012 MQT: Connect failed to mosquitto1:1883, rc -5. Retry in 10 sec
20:31:23.390 CFG: Saved to flash at F5, Count 1062, Bytes 4096
20:31:28.555 WIF: Checking connection...
20:31:28.557 WIF: Attempting connection...
20:31:29.567 WIF: Connecting to AP2 VictoriaDrive3 Channel 5 BSSId 1C:61:B4:6B:CC:F6 in mode 11n as AquariumLights-2458...
20:31:30.751 WIF: Checking connection...
20:31:30.753 WIF: Attempting connection...
20:31:31.752 WIF: Checking connection...
20:31:31.754 WIF: Attempting connection...
20:31:32.754 WIF: Checking connection...
20:31:32.756 WIF: Attempting connection...
20:31:33.758 WIF: Checking connection...
20:31:33.760 WIF: Attempting connection...
20:31:34.762 WIF: Checking connection...
20:31:34.763 WIF: Attempting connection...
20:31:35.763 WIF: Checking connection...
20:31:35.764 WIF: Attempting connection...
20:31:36.764 WIF: Checking connection...
20:31:36.766 WIF: Connect failed
20:31:37.774 WIF: Connecting to AP1 VictoriaDrive2B in mode 11n as AquariumLights-2458...
20:31:38.917 CFG: Saved to flash at F4, Count 1063, Bytes 4096
20:31:39.038 WIF: Checking connection...
20:31:39.040 WIF: Attempting connection...
20:31:40.504 WIF: Checking connection...
20:31:40.506 WIF: Connected
20:31:40.854 WIF: Sending Gratuitous ARP
20:31:41.374 CFG: Saved to flash at FB, Count 1064, Bytes 4096
20:31:51.025 MQT: Attempting connection...
20:31:52.029 WIF: DNS failed for mosquitto1 after 1001 ms
20:31:52.032 MQT: Connect failed to mosquitto1:1883, rc -5. Retry in 20 sec
20:32:02.056 WIF: Checking connection...
20:32:13.829 MQT: Attempting connection...
20:32:13.851 WIF: DNS resolved 'mosquitto1' (192.168.9.1) in 19 ms
20:32:13.901 MQT: Connected
20:32:13.905 MQT: tele/AquariumLights/LWT = Online (retained)
20:32:13.908 MQT: cmnd/AquariumLights/POWER = 
20:32:13.911 MQT: Subscribe to cmnd/AquariumLights/#
20:32:13.914 MQT: Subscribe to cmnd/tasmotas/#
20:32:13.917 MQT: Subscribe to cmnd/DVES_AC899A_fb/#
20:32:13.939 MQT: tasmota/discovery/600194AC899A/config = {"ip":"192.168.9.27","dn":"Aquarium Lights","fn":["Aquarium_Lights_1","Aquarium_Lights_2",null,null,null,null,null,null],"hn":"AquariumLights-2458","mac":"600194AC899A","md":"Sonoff Dual R2","ty":0,"if":0,"ofln":"Offline","onln":"Online","state":["OFF","ON","TOGGLE","HOLD"],"sw":"14.3.0","t":"AquariumLights","ft":"%prefix%/%topic%/","tp":["cmnd","stat","tele"],"rl":[1,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],"swc":[-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1],"swn":[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],"btn":[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],"so":{"4":0,"11":0,"13":0,"17":0,"20":0,"30":0,"68":0,"73":0,"82":0,"114":0,"117":0},"lk":0,"lt_st":0,"bat":0,"dslp":0,"sho":[],"sht":[],"ver":1} (retained)
20:32:13.948 MQT: tasmota/discovery/600194AC899A/sensors = {"sn":{"Time":"2024-12-19T20:32:13"},"ver":1} (retained)
20:32:14.306 SRC: MQTT
20:32:14.309 CMD: Grp 0, Cmd 'POWER', Idx 1, Len 2, Pld 1, Data 'on'
20:32:14.316 MQT: stat/AquariumLights/RESULT = {"POWER1":"ON"}
20:32:14.320 MQT: stat/AquariumLights/POWER1 = ON
20:32:14.357 SRC: MQTT
20:32:14.360 CMD: Grp 0, Cmd 'POWER', Idx 2, Len 3, Pld 0, Data 'off'
20:32:14.365 MQT: stat/AquariumLights/RESULT = {"POWER2":"OFF"}
20:32:14.368 MQT: stat/AquariumLights/POWER2 = OFF
20:32:22.575 WIF: Checking connection...
```
Of course, being daytime, it didn't toggle...
10 hours to go before it's "supposed" to be in night mode...

**Reply by Jason2866** [reply]
Have you tried replace the hardware. esp8266 and if an extra power supply the power supply too.
Borderline (dying) hardware can make all sort of weird issues.

**Reply by cdntinker** [reply]
That would definitely be a pain.  Apparently, Sonoff has discontinued the Dual R2...

**Reply by sfromis** [reply]
They have the Sonoff Dual R3 instead, and -lite without energy monitoring.

Recently, I replaced an old Sonoff Pow R2, due to "bad behavior", suddenly started to click the relay without me doing anything.

**Reply by cdntinker** [reply]
Which requires a redesign of the control box setup & awkwardly flashing the device. (Why on earth has Sonoff gone away from properly laying out those pads???)

**Reply by cdntinker** [reply, marked as answer]
So...
In preparation for replacement of the Dual R2 with a Dual R3...
I dug out a Dual R3 that I had completely forgotten buying.
Yay!  I had already flashed it.
Updated it to 14.4.1...
Took the R2 offline while setting up the R3 to confirm it would act properly as a replacement...
Gave the R3 the WiFi & MQTT configuration of the R2...
Powered it up...

And...

SAME DAMN ISSUE!


Which told me that the problem was NOT the device itself.


But!!!

Now for the good news.

Re started mosquitto_sub sessions in 2 terminals & noticed something odd.

Renamed the R3 & put the R2 back online.

I have NO idea how it happened, but somehow, (& this is the weird part...)  

When I started the mosquitto_sub for AquariumLights:

cmnd/AquariumLights/Power1 on
and
cmnd/AquariumLights/Power2 off

showed immediately...

Apparently, those had somehow been retained on both the regular broker and my test broker.

Clearing those 2 retained messages from the brokers seems to have fixed it.

**Reply by sfromis** [reply]
Such things are why I suggested you to look at relevant and detailed logs, at the relevant point in time. Very important to debugging.

**Reply by cdntinker** [reply]
And yet, the logs didn't point out anything when the issue did occur.
The only thing that showed up the problem was a chance viewing of MQTT traffic side-by-side with a non-broken equivalent.

**Reply by sfromis** [reply]
You never posted the detailed logs requested (even after two reminders), which is where clues would've been, including that commands arrived via MQTT.

**Reply by cdntinker** [reply]
Not sure which detailed logs I missed.  I did eventually find the output during the issue from weblog 4 & post it.

**Reply by sfromis** [reply]
What I meant was answering the question
> Set `weblog` to 4 and then, **when you experience your issue**, provide the output of the Console log

When you posted some logs from the relevant time, it was without `Weblog 4` activated, and when you then activated `Weblog 4`, it was at a point in time where nothing happened, hence nothing to see.

Not trying to make a big case out of it, more like pointing out that debugging takes getting the details.

From the Tasmota FAQ....
https://tasmota.github.io/docs/FAQ/#my-device-randomly-switches-on-and-off-do-i-have-ghosts-in-my-house

**Reply by cdntinker** [reply]
Well... I did eventually catch it doing the reconnect & posted that section of the log too.  So I'm at a loss as to what further logs you wanted to see...

**Reply by sfromis** [reply]
Since you concluded with "Of course, being daytime, it didn't toggle...", that suggested not much to look for there, not being "when you experience the issue".

Anyway, it does have the strong clue:
> `20:32:14.306 SRC: MQTT`

This indicates that commands came in via MQTT, which would be either some automation, or retained messages (which it was).
